### DISCUSSION TITLE
Game crashes on input

### DISCUSSION THREAD

**Discussion by 3nderall** [stride3d/stride#1861] [Q&A] [answered]
I just started developing in stride with only basic unity knowledge. I managed to get my character jumping (I'm making a flappy bird clone) and when I press space a cmd window opens and then the game window and the cmd window closes. Why is this happening?
Here is my script:
```
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Stride.Core.Mathematics;
using Stride.Input;
using Stride.Engine;
using BulletSharp;

namespace flappyBord
{
    public class birdScript : SyncScript
    {
        public float jumpPower;
        public RigidBody rigidBody;

        public override void Start()
        {
            
        }

        public override void Update()
        {
            

            if (Input.IsKeyDown(Keys.Space))
            {
                rigidBody.LinearVelocity = new Vector3(0, jumpPower, 0);
            }
        }
    }
}

```

**Comment by 3nderall** [comment]
I can see my character jumping before the game crashes

**Comment by Doprez** [comment]
if you press the run button in VS 2022 it should also give you a better error.

You may also prefer Input.IsKeyPressed instead of Input.IsKeyDown here as that will only trigger once per button press.

**Reply by 3nderall** [reply]
I ran it in VS and the problem is the rigidBody variable isn't populated but I can't figure out how to populate it. In unity, you can just drag the rigid body module into the empty field.

**Reply by Doprez** [reply]
can you try `public RigidBody rigidBody{ get; set; };` instead of `public RigidBody rigidBody;` and then assign it in editor? 

If the Rigidbody is attached to the Entity you can also use Entity.Get<RigidBody>() to find the component in the same entity and assign it at Start().

**Reply by 3nderall** [reply]
> can you try `public RigidBody rigidBody{ get; set; };` instead of `public RigidBody rigidBody;` and then assign it in editor?
> 
> If the Rigidbody is attached to the Entity you can also use Entity.Get() to find the component in the same entity and assign it at Start().

I just came off of my computer but I'll try this tomorrow.

**Reply by 3nderall** [reply]
That still doesn't work

**Comment by Eideren** [comment, marked as answer]
That's the wrong `Rigidbody` type, you're using bullet's. Remove your `using BulletSharp;` and change `public RigidBody rigidBody;` to `public RigidBodyComponent rigidBody` instead

**Reply by 3nderall** [reply]
Thanks for the solution, the the body in RigidBodyComponent is a lowercase b instead though.
